### ISSUE TITLE
unhandled exception when parsing VCard's "REV" field

### ISSUE THREAD

**Issue by oli-arborum** [Kozea/Radicale#829]
Installed Radicale 2.1.9 in a Docker container using the provided Dockerfile under Ubuntu Server 16.04.4 LTS.
Using OS X 10.9.5's Contacts app to copy vcards from my main address book to Radicale's.
I always get a "400 bad request" due to an unhandled exception when parsing the contents of the "REV" field, see debug output:
```
[7f5c31f28ae8] INFO: PUT request for '/oli/7e1d737b-8190-45a0-4b40-b05edc3efe60/6641bc09-5d19-4092-b8e6-2860c0c25ea8.vcf' received from 192.168.178.22 using 'Mac OS X/10.9.5 (13F1911) AddressBook/1371.2'
[7f5c31f28ae8] DEBUG: Request headers:
{'CONTENT_LENGTH': '598',
 'CONTENT_TYPE': 'text/vcard; charset=utf-8',
 'GATEWAY_INTERFACE': 'CGI/1.1',
 'HTTP_ACCEPT': '*/*',
 'HTTP_ACCEPT_ENCODING': 'gzip, deflate',
 'HTTP_ACCEPT_LANGUAGE': 'de-de',
 'HTTP_CONNECTION': 'keep-alive',
 'HTTP_HOST': 'radicale.192.168.178.68.xip.io:5232',
 'HTTP_IF_NONE_MATCH': '*',
 'HTTP_USER_AGENT': 'Mac OS X/10.9.5 (13F1911) AddressBook/1371.2',
 'PATH_INFO': '/oli/7e1d737b-8190-45a0-4b40-b05edc3efe60/6641bc09-5d19-4092-b8e6-2860c0c25ea8.vcf',
 'QUERY_STRING': '',
 'REMOTE_ADDR': '192.168.178.22',
 'REMOTE_HOST': '',
 'REQUEST_METHOD': 'PUT',
 'SCRIPT_NAME': '',
 'SERVER_NAME': 'ae822c5bc6e7',
 'SERVER_PORT': '5232',
 'SERVER_PROTOCOL': 'HTTP/1.1',
 'SERVER_SOFTWARE': 'WSGIServer/0.2',
 'wsgi.errors': <_io.StringIO object at 0x7f5c32bb4558>,
 'wsgi.file_wrapper': <class 'wsgiref.util.FileWrapper'>,
 'wsgi.input': <_io.BufferedReader name=6>,
 'wsgi.multiprocess': False,
 'wsgi.multithread': True,
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'http',
 'wsgi.version': (1, 0)}
[7f5c31f28ae8] DEBUG: Sanitized script name: ''
[7f5c31f28ae8] DEBUG: Sanitized path: '/oli/7e1d737b-8190-45a0-4b40-b05edc3efe60/6641bc09-5d19-4092-b8e6-2860c0c25ea8.vcf'
[7f5c31f28ae8] DEBUG: Request content:
BEGIN:VCARD
VERSION:3.0
PRODID:-//Apple Inc.//Mac OS X 10.9.5//EN
N:_Mustermann;_Max;;Dipl.-Ing.;
FN:Dipl.-Ing. _Max _Mustermann
ORG:Firma;
EMAIL;type=INTERNET;type=HOME;type=pref:mail@mailadresse.de
EMAIL;type=INTERNET;type=WORK:User1 (mail2)
TEL;type=CELL;type=VOICE;type=pref:+49-171-999999
TEL;type=WORK;type=VOICE:+49-999-99999
ADR;type=HOME;type=pref:;;Privatstraße;Privatort;Privatbundesland;99998;Privatland
ADR;type=WORK:;;Bürostraße 42;Büroort;Bürobundesland;99999;Büroland
BDAY:1975-11-22
REV:2016-10-01T18:11:36Z
UID:6641bc09-5d19-4092-b8e6-2860c0c25ea8
END:VCARD

[7f5c31f28ae8] WARNING: Bad PUT request on '/oli/7e1d737b-8190-45a0-4b40-b05edc3efe60/6641bc09-5d19-4092-b8e6-2860c0c25ea8.vcf': At line 14: In transformToNative, unhandled exception on line 14: <class 'ValueError'>: invalid literal for int() with base 10: '0-' (<REV{}2016-10-01T18:11:36Z>)
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/radicale_vobject/icalendar.py", line 1733, in stringToDateTime
    day = int(s[6:8])
ValueError: invalid literal for int() with base 10: '0-'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/radicale_vobject/icalendar.py", line 1956, in parseDtstart
    return stringToDateTime(contentline.value, tzinfo)
  File "/usr/lib/python3.6/site-packages/radicale_vobject/icalendar.py", line 1741, in stringToDateTime
    raise ParseError("'{0!s}' is not a valid DATE-TIME".format(s))
radicale_vobject.base.ParseError: "'2016-10-01T18:11:36Z' is not a valid DATE-TIME"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/radicale_vobject/base.py", line 185, in transformToNative
    return self.behavior.transformToNative(self)
  File "/usr/lib/python3.6/site-packages/radicale_vobject/icalendar.py", line 808, in transformToNative
    obj.value = parseDtstart(obj, allowSignatureMismatch=True)
  File "/usr/lib/python3.6/site-packages/radicale_vobject/icalendar.py", line 1959, in parseDtstart
    return stringToDate(contentline.value)
  File "/usr/lib/python3.6/site-packages/radicale_vobject/icalendar.py", line 1722, in stringToDate
    day = int(s[6:8])
ValueError: invalid literal for int() with base 10: '0-'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/radicale/__init__.py", line 854, in do_PUT
    items = tuple(vobject.readComponents(content or ""))
  File "/usr/lib/python3.6/site-packages/radicale_vobject/base.py", line 1125, in readComponents
    component.transformChildrenToNative()
  File "/usr/lib/python3.6/site-packages/radicale_vobject/base.py", line 673, in transformChildrenToNative
    child = child.transformToNative()
  File "/usr/lib/python3.6/site-packages/radicale_vobject/base.py", line 198, in transformToNative
    raise ParseError(msg, lineNumber)
radicale_vobject.base.ParseError: At line 14: In transformToNative, unhandled exception on line 14: <class 'ValueError'>: invalid literal for int() with base 10: '0-' (<REV{}2016-10-01T18:11:36Z>)
[7f5c31f28ae8] DEBUG: Response content:
Bad Request
[7f5c31f28ae8] INFO: PUT response status for '/oli/7e1d737b-8190-45a0-4b40-b05edc3efe60/6641bc09-5d19-4092-b8e6-2860c0c25ea8.vcf' in 0.005 seconds: 400 Bad Request
```

**Comment by Unrud** [comment]
It's a bug in the development version of VObject (https://github.com/eventable/vobject/issues/120). Use a stable release of Radicale and not the version from the master branch.

**Comment by oli-arborum** [comment]
Sorry, I downloaded the stable 2.1.9 and didn't see, that the Dockerfile downloads the head of the master branch.
Maybe this should also be fixed?

**Comment by enkore** [comment]
The mentioned bug was released in vobject 0.9.6 a few days ago, but a fixed 0.9.6.1 has just been released.
